File: assignments/AS01/rps.py
import numbers
import random

numGames = int(input("best of?"))
playerScore = 0
comScore = 0

gameOn = True;

options = ["rock", "paper", "scissors"]

while gameOn:
  randNum = random.randint(0,2)
  comChoice = options[randNum]
  playerChoice = input("choose rock, paper, or scissors: ")

  if playerChoice == comChoice:
    print("tie")
  elif playerChoice == "rock":
    if comChoice == "scissors":
      playerScore += 1
      print("W")
    else:
      print("L")
      comScore += 1
  elif playerChoice == "paper":
    if comChoice == "rock":
      playerScore += 1
      print("W")
    else:
      print("L")
      comScore += 1
  elif playerChoice == "scissors":
    if comChoice == "paper":
      playerScore += 1
      print("W")
    else:
      print("L")
      comScore += 1

  print("player's score: " + str(playerScore) + " , computer's score: " + str(computerScore))

  if playerScore > numGames:
    print("player wins! GG")
    gameOver = True
  elif comScore > numGames:
    print("computer wins! GG")
    gameOver = True

File: assignments/AS01/rps.cpp
#include <time.h>

#include <cstdlib>
#include <iostream>

// 0 == tie; 1 == player; -1 == cpu;
int check_conditions(std::string player, std::string cpu) {
  if (player == cpu) return 0;
  if ((player == "rock" && cpu == "scissors") ||
      (player == "paper" && cpu == "rock") ||
      (player == "scissors" && cpu == "paper"))
    return 1;
  return -1;
}

int main() {
  srand(time(0));

  int num_games;

  std::cout << "best of?\n";
  std::cin >> num_games;

  bool is_playing = true;
  std::string options[] = {"rock", "paper", "scissors"};

  int player_score = 0;
  int cpu_score = 0;

  while (is_playing) {
    std::string cpu_choice = options[rand() % 3];
    std::string player_choice;
    std::cout << "rock, paper, or scissors?\n";
    std::cin >> player_choice;
    int winner = check_conditions(player_choice, cpu_choice);
    if (winner == 1) {
      player_score++;
      std::cout << "player wins round!\n";
    } else if (winner == -1) {
      cpu_score++;
      std::cout << "cpu wins round!\n";
    } else {
      std::cout << "tied round\n";
    }
    std::cout << "player's score: " << player_score
              << ", computer's score: " << cpu_score << '\n';
    if (player_score == num_games / 2 + 1) {
      std::cout << "player wins! GG\n";
      is_playing = false;
    } else if (cpu_score == num_games / 2 + 1) {
      std::cout << "cpu wins! GG\n";
      is_playing = false;
    }
  }
  return 0;
}
